Library/Homebrew/formula_installer.rb

```diff
@@ -589,6 +589,8 @@ def build
       end
     end
 
+    formula.update_head_version
```

Comment:
Besides this place. You also need to update head version in clean logic https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula_installer.rb#L599